perception/compare_map_segmentation/include/compare_map_segmentation/voxel_grid_map_loader.hpp

```diff
@@ -0,0 +1,219 @@
+    }
+    for (size_t i = 0; i < map_cell_ids_to_remove.size(); ++i) {
+      removeMapCell(map_cell_ids_to_remove.at(i));
+    }
+  }
+
+  inline void removeMapCell(const std::string map_cell_id_to_remove)
+  {
+    (*mutex_ptr_).lock();
+    current_voxel_grid_dict_.erase(map_cell_id_to_remove);
+    (*mutex_ptr_).unlock();
+  }
+
+  inline void addMapCellAndFilter(
+    const autoware_map_msgs::msg::PointCloudMapCellWithID & map_cell_to_add)
+  {
+    auto exe_start_time = std::chrono::system_clock::now();
+
+    pcl::PointCloud<pcl::PointXYZ> map_cell_pc_tmp;
+    pcl::fromROSMsg(map_cell_to_add.pointcloud, map_cell_pc_tmp);
+
+    MultiVoxelGrid map_cell_voxel_grid_tmp;
+    PointCloudPtr map_cell_downsampled_pc_ptr_tmp;
+
+    auto map_cell_voxel_input_tmp_ptr =
+      std::make_shared<pcl::PointCloud<pcl::PointXYZ>>(map_cell_pc_tmp);
+    map_cell_voxel_grid_tmp.setLeafSize(voxel_leaf_size_, voxel_leaf_size_, voxel_leaf_size_);
+    map_cell_downsampled_pc_ptr_tmp.reset(new pcl::PointCloud<pcl::PointXYZ>);
+    map_cell_voxel_grid_tmp.setInputCloud(map_cell_voxel_input_tmp_ptr);
+    map_cell_voxel_grid_tmp.setSaveLeafLayout(true);
+    map_cell_voxel_grid_tmp.filter(*map_cell_downsampled_pc_ptr_tmp);
+
+    MapGridVoxelInfo current_voxel_grid_list_item;
+    // TODO(badai-nguyen): use map cell info from map cell, when map loader I/F is updated
+    current_voxel_grid_list_item.min_b_x = map_cell_voxel_grid_tmp.get_min_p()[0];
+    current_voxel_grid_list_item.min_b_y = map_cell_voxel_grid_tmp.get_min_p()[1];
+    current_voxel_grid_list_item.max_b_x = map_cell_voxel_grid_tmp.get_max_p()[0];
+    current_voxel_grid_list_item.max_b_y = map_cell_voxel_grid_tmp.get_max_p()[1];
+
+    current_voxel_grid_list_item.map_cell_voxel_grid.set_voxel_grid(
+      &(map_cell_voxel_grid_tmp.leaf_layout_), map_cell_voxel_grid_tmp.get_min_b(),
+      map_cell_voxel_grid_tmp.get_max_b(), map_cell_voxel_grid_tmp.get_div_b(),
+      map_cell_voxel_grid_tmp.get_divb_mul(), map_cell_voxel_grid_tmp.get_inverse_leaf_size());
+
+    current_voxel_grid_list_item.map_cell_pc_ptr.reset(new pcl::PointCloud<pcl::PointXYZ>);
+    // for (size_t i = 0; i < map_cell_downsampled_pc_ptr_tmp->points.size(); ++i) {
+    //   current_voxel_grid_list_item.map_cell_pc_ptr->points.push_back(
+    //     map_cell_downsampled_pc_ptr_tmp->points.at(i));
+    // }
+    current_voxel_grid_list_item.map_cell_pc_ptr = std::move(map_cell_downsampled_pc_ptr_tmp);
+    // add
+    (*mutex_ptr_).lock();
+    current_voxel_grid_dict_.insert({map_cell_to_add.cell_id, current_voxel_grid_list_item});
+    (*mutex_ptr_).unlock();
+
+    auto exe_stop_time = std::chrono::system_clock::now();
+    const double exe_run_time =
+      std::chrono::duration_cast<std::chrono::microseconds>(exe_stop_time - exe_start_time)
+        .count() /
+      1000.0;
+    RCLCPP_INFO(
+      logger_, " map cell %s adding time: %lf", map_cell_to_add.cell_id.c_str(), exe_run_time);
```

Comment:
Does this print every time?

Reply:
For each time a map cell is loaded and added to buffer. It might be useful to decide `timer_interval_ms` param value. Anyway it is no problem if remove it.

Reply:
If unnecessary logs flow, the output of the terminal will be chaotic. If it is unnecessary, it may be better to delete it or use ROSCPP_DEBUG. :+1:

Reply:
I remove it by [this](https://github.com/autowarefoundation/autoware.universe/pull/3087/commits/7126fbd899ea3e86a49c7ca9717e196be3ec92a1)